core/src/main/java/com/linecorp/armeria/server/AbstractServiceBindingBuilder.java

```diff
@@ -218,35 +118,10 @@ final void build0(Service<HttpRequest, HttpResponse> service) {
         final List<Route> routes = buildRouteList();
 
         for (Route route : routes) {
-            final ServiceConfigBuilder serviceConfigBuilder =
-                    new ServiceConfigBuilder(route, decorate(service));
-            if (requestTimeoutMillis != null) {
-                serviceConfigBuilder.requestTimeoutMillis(requestTimeoutMillis);
-            }
-            if (maxRequestLength != null) {
-                serviceConfigBuilder.maxRequestLength(maxRequestLength);
-            }
-            if (verboseResponses != null) {
-                serviceConfigBuilder.verboseResponses(verboseResponses);
-            }
-            if (requestContentPreviewerFactory != null) {
-                serviceConfigBuilder.requestContentPreviewerFactory(requestContentPreviewerFactory);
-            }
-            if (responseContentPreviewerFactory != null) {
-                serviceConfigBuilder.responseContentPreviewerFactory(responseContentPreviewerFactory);
-            }
-            if (accessLogWriter != null) {
-                serviceConfigBuilder.accessLogWriter(accessLogWriter, shutdownAccessLogWriterOnStop);
-            }
+            service = defaultServiceConfigSetters.decorate(service);
+            final ServiceConfigBuilder serviceConfigBuilder = defaultServiceConfigSetters
+                                                                    .toServiceConfigBuilder(route, service);
```

Comment:
I think we need a new local variable here unless decorating a service multiple times.
```suggestion
            final decoratedService = defaultServiceConfigSetters.decorate(service);
            final ServiceConfigBuilder serviceConfigBuilder = defaultServiceConfigSetters
                                                                    .toServiceConfigBuilder(route, decoratedService);
```